File: hashira.js
let input = "";
process.stdin.on("data", (chunk) => (input += chunk));
process.stdin.on("end", () => {
  // Parse JSON
  const data = JSON.parse(input);
  const n = data.keys.n;
  const k = data.keys.k;

  // Helper to convert string in given base to decimal (BigInt)
  function baseToDecimal(value, base) {
    return BigInt(parseInt(value, base));
  }
  let roots = [];
  for (let i = 1; i <= k; ++i) {
    const root = data[i.toString()];
    if (!root) continue;
    roots.push(baseToDecimal(root.value, parseInt(root.base)));
  }

  if (roots.length < k) {
    console.log("Not enough roots found in input.");
  } else {
    // For degree m = k-1: constant term is (-1)^k * r1*r2*...*rk
    let c = BigInt(1);
    for (let i = 0; i < k; ++i) c *= roots[i];
    if (k % 2 === 1) c = -c;
    console.log("The value of c is: " + c.toString());
  }
});

File: Hashira.cpp
#include <iostream>
#include <string>
#include <vector>
#include <map>
#include <sstream>
#include <cctype>
#include <iomanip>
using namespace std;

// Convert a string in given base to decimal
long long convertToDecimal(const string &s, int base) {
    long long val = 0;
    for (char c : s) {
        int digit;
        if (isdigit(c)) digit = c - '0';
        else if (isalpha(c)) digit = 10 + (tolower(c) - 'a');
        else continue;
        val = val * base + digit;
    }
    return val;
}

// Lagrange interpolation → only constant term
long double lagrangeConstant(const vector<long double> &x, const vector<long double> &y) {
    int n = x.size();
    long double constantTerm = 0.0;

    for (int i = 0; i < n; i++) {
        long double term = y[i];
        long double denom = 1.0;
        for (int j = 0; j < n; j++) {
            if (i == j) continue;
            term *= (-x[j]);
            denom *= (x[i] - x[j]);
        }
        constantTerm += term / denom;
    }
    return constantTerm;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    // Read JSON input
    string input, line;
    while (getline(cin, line)) {
        input += line;
    }

    // Extract n and k
    size_t posN = input.find("\"n\"");
    size_t posK = input.find("\"k\"");
    int n = stoi(input.substr(input.find(":", posN)+1));
    int k = stoi(input.substr(input.find(":", posK)+1));

    vector<long double> x, y;
    int count = 0;

    // Parse roots until we get k values
    for (int idx = 1; idx <= n && count < k; idx++) {
        string key = "\"" + to_string(idx) + "\"";
        size_t pos = input.find(key);
        if (pos == string::npos) continue;

        // Base
        size_t posBase = input.find("\"base\"", pos);
        size_t startBase = input.find("\"", posBase+7) + 1;
        size_t endBase = input.find("\"", startBase);
        int base = stoi(input.substr(startBase, endBase - startBase));

        // Value
        size_t posVal = input.find("\"value\"", pos);
        size_t start = input.find("\"", posVal+7) + 1;
        size_t end = input.find("\"", start);
        string valStr = input.substr(start, end - start);

        long long val = convertToDecimal(valStr, base);

        x.push_back(idx); // root index
        y.push_back(val); // converted value
        count++;
    }

    // Compute constant term
    long double c = lagrangeConstant(x, y);

   cout << "Constant term (c) = " << c << "\n";
    return 0;
}
